File: Assets/XLuaExamples/Menu.cs
/* Author:          ezhex1991@outlook.com
 * CreateTime:      2018-10-11 13:14:44
 * Organization:    #ORGANIZATION#
 * Description:     
 */
using EZhex1991.EZUnity.Framework;
using UnityEngine;
using UnityEngine.SceneManagement;
using UnityEngine.UI;

namespace EZhex1991.EZUnity.XLuaExample
{
    public class Menu : MonoBehaviour
    {
        public Button button_Home;
        public Button button_LuaBehaviour;
        public Button button_LuckyBall;
        public Button button_SpaceShooter;

        public GameObject panel_Menu;

        public void Awake()
        {
            if (EZApplication.Instance.runMode == RunMode.Update && !EZResources.Instance.isUpdated)
            {
                panel_Menu.SetActive(false);
                EZResources.Instance.onUpdateCompleteEvent += () =>
                {
                    panel_Menu.SetActive(true);
                };
            }
            button_Home.onClick.AddListener(LoadMenu);
            button_LuaBehaviour.onClick.AddListener(LoadLuaBehaviour);
            button_LuckyBall.onClick.AddListener(LoadLuckyBall);
            button_SpaceShooter.onClick.AddListener(LoadSpaceShooter);
        }

        public void LoadMenu()
        {
            SceneManager.LoadScene("Menu", LoadSceneMode.Single);
        }

        public void LoadLuaBehaviour()
        {
            SceneManager.LoadScene("LuaBehaviour", LoadSceneMode.Additive);
            panel_Menu.SetActive(false);
        }
        public void LoadLuckyBall()
        {
            SceneManager.LoadScene("LuckyBall", LoadSceneMode.Additive);
            panel_Menu.SetActive(false);
        }
        public void LoadSpaceShooter()
        {
            EZResources.Instance.LoadSceneAsync("spaceshooter", "SpaceShooter", LoadSceneMode.Additive);
            panel_Menu.SetActive(false);
        }
    }
}

File: Assets/XLuaExamples/Script_Lua/Menu.lua
--[==[
- Author:       ezhex1991@outlook.com
- CreateTime:   2018-02-27 17:18:49
- Orgnization:  #ORGNIZATION#
- Description:  
--]==]
local M = {}
----- CODE -----
local LoadSceneMode = CS.UnityEngine.SceneManagement.LoadSceneMode

function M.LuaAwake(injector)
    M.btns = {}
    injector:Inject(M)
    M.btns[1].onClick:AddListener(M.Menu)
    M.btns[2].onClick:AddListener(M.LuaBehaviour)
    M.btns[3].onClick:AddListener(M.LuckyBall)
    M.btns[4].onClick:AddListener(M.SpaceShooter)
    return M
end

function M.Menu()
    CS.UnityEngine.SceneManagement.SceneManager.LoadScene("Menu")
end

function M.LuaBehaviour()
    CS.UnityEngine.SceneManagement.SceneManager.LoadScene("LuaBehaviour", LoadSceneMode.Additive)
    M.panel_Menu:SetActive(false)
end

function M.LuckyBall()
    CS.UnityEngine.SceneManagement.SceneManager.LoadScene("LuckyBall", LoadSceneMode.Additive)
    M.panel_Menu:SetActive(false)
end

function M.SpaceShooter()
    require("SpaceShooter.GameController"):Init()
    M.panel_Menu:SetActive(false)
end
----- CODE -----
return M
